wandb/sdk/wandb_settings.py

```diff
@@ -957,12 +957,10 @@ def load(self, fname: str) -> None:
         pass
 
     def __setattr__(self, name: str, value: Any) -> None:
-        if name not in self.__dict__:
-            raise AttributeError(name)
-        if self.__frozen:
-            raise TypeError("Settings object is frozen")
-        value = self._perform_preprocess(name, value)
-        self._check_invalid(name, value)
+        try:
+            self._update({name: value}, _source=self.Source.SETUP)
```

Comment:
im nervous about this.  Setup is meant to be settings configured from:
```
wandb.setup(settings=)
```
I think the real problem is that we arent applying SOURCE.INIT to un sourced settings in apply_init()
this is a mess. sorry.  let me see if I can propose something better or @KyleGoyette might have some ideas.

Reply:
Yeah, I was a bit nervous about this too.  Setting this to `self.Source.SETTINGS` broke a few tests.  It's really only exposed in our tests because we construct a settings object and then mutate it, which might be an anti-pattern we could just remove.  The workflow is:

```
settings = wandb.Settings(console="off")
settings.ignore_globs = ["foo"]
wandb.init(settings=settings)
```

Reply:
I'm also nervous about this change. I don't think it will immediately raise issues, but it will be confusing later. Could we instead have the test_settings fixture accept some arguments to directly set the settings when it is instantiated? Or at least add a comment indicating why we do this, so when I get around to the settings rework, I have some context to work from?